frappe/public/js/frappe/form/controls/datetime.js

```diff
@@ -81,9 +81,6 @@ frappe.ui.form.ControlDatetime = class ControlDatetime extends frappe.ui.form.Co
 
 	get_model_value() {
 		let value = super.get_model_value();
-		if (!value && !this.doc) {
-			value = this.last_value;
-		}
```

Comment:
This is no longer needed, since `get_model_value` return `this.value` as fallback now.